src/ScottPlot5/ScottPlot5/Primitives/SvgImage.cs

```diff
@@ -17,9 +19,30 @@ public SvgImage(int width, int height)
         Canvas = SKSvgCanvas.Create(rect, Stream);
     }
 
+    private static int CountSubstring(string text, string substring)
+    {
+#if NET5_0_OR_GREATER
+        int count = Regex.Matches(text, substring).Count;
+
+#else
+        var expression = new Regex(substring);
+        int count = expression.Matches(text).Count;
+#endif
+        return count;
+    }
+
+    private static string[] SvgTags = new string[] { "g", "svg" };
```

Comment:
Do we know that this issue only happens for these tags? Or preferably why it happens, so we can hopefully fix it in a better way? We already cannot assume that `<svg>` is never contained within a `<g>` tag (which this code implicitly does), as you are allowed to nest SVG documents inside eachother. This would get much worse if we had to add more tags.

It would be very helpful to see some generated documents and where they go wrong, because it's entirely possible that this is exactly enough. But right now I'm seeing that it only appends two closing tags (and only appends each closing tag at most once), which seems like it's either not enough, or it indicates some very specific issue, such as not properly telling Skia that we want to terminate the document.

In fact, I'm seeing the following remark in the Skiasharp docs that makes me think I'm right and we can make Skia close the document itself: https://learn.microsoft.com/en-us/dotnet/api/skiasharp.sksvgcanvas?view=skiasharp-2.88#remarks
```
The canvas may buffer some drawing calls, so the output is not guaranteed to be valid or complete until the canvas instance is deleted.
```

I assume by deleting the canvas instance they mean disposing it, as I don't see any API that would be a better fit. Not the wording I would expect in a C# API, but it's probably just borrowed from the underlying C++ API where that's a more common term to use. Fortunately, the `SvgImage` class is marked `internal` and I believe never exposed to users, so I this wouldn't be a breaking change.

Reply:
I'm seeing that the following patch properly closes the image without needing to append a `</svg>`. I think this is probably the way to go, but we should probably have some logic to throw if they call `GetXml` twice.

```
diff --git a/src/ScottPlot5/ScottPlot5/Primitives/SvgImage.cs b/src/ScottPlot5/ScottPlot5/Primitives/SvgImage.cs
index 03edcc1a3..0b53dfc64 100644
--- a/src/ScottPlot5/ScottPlot5/Primitives/SvgImage.cs
+++ b/src/ScottPlot5/ScottPlot5/Primitives/SvgImage.cs
@@ -19,7 +19,8 @@ internal class SvgImage : IDisposable

     public string GetXml()
     {
-        return Encoding.UTF8.GetString(Stream.ToArray()) + "</svg>";
+        Canvas.Dispose();
+        return Encoding.UTF8.GetString(Stream.ToArray());
     }

     public void Dispose()
```

As an aside, it's also very C++-ey to document object destruction particulars and key parts of using the API as a small remark rather than in a more conspicuous area.